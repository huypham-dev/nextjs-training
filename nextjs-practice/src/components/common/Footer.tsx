import { memo } from "react";
import Link from "next/link";

const Footer = (): JSX.Element => (
  <footer className="w-full bg-white px-6 border-t">
    <div className="container mx-auto max-w-4xl py-6 flex flex-wrap md:flex-no-wrap justify-between items-center text-sm">
      &copy;2021 Your Company. All rights reserved.
      <div className="pt-4 md:p-0 text-center md:text-right text-xs">
        <Link href="/policy">
          <a className="text-black no-underline hover:underline">
            Privacy Policy
          </a>
        </Link>
        <Link href="/condition">
          <a className="text-black no-underline hover:underline ml-4">
            Terms &amp; Conditions
          </a>
        </Link>
        <Link href="/contact">
          <a className="text-black no-underline hover:underline ml-4">
            Contact Us
          </a>
        </Link>
      </div>
    </div>
  </footer>
);

export default memo(Footer);
