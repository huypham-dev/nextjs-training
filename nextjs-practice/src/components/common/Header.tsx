import { memo } from "react";
import Image from "next/image";
import Link from "next/link";
import { NavigationItem } from "types";
import isEqual from "react-fast-compare";

type Props = {
  navigations: NavigationItem[];
};

const Header = ({ navigations }: Props) => (
  <header className="w-full mx-auto text-center">
    <Link href="/">
      <a className="inline-block text-2xl mb-12">
        <Image
          src="https://images.prismic.io/nextjsexample/ebd683dd-af14-49e3-bef5-5c497c9b8cac_logo.png"
          alt="logo"
          width={150}
          height={150}
        />
      </a>
    </Link>

    <ul className="flex justify-center b p-5 uppercase text-xs border-t border-b bg-gray-100">
      {navigations.map(({ label, path }) => (
        <li key={label}>
          <Link href={path}>
            <a className="text-gray-600 font-bold text-sm py-2 px-4 mx-2 hover:text-black">
              {label}
            </a>
          </Link>
        </li>
      ))}
    </ul>
  </header>
);

export default memo(Header, isEqual);
