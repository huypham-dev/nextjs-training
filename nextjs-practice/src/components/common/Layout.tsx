import { navigations } from "constants/mockdata";
import Footer from "./Footer";
import Header from "./Header";

type Props = {
  children: React.ReactNode;
};

const Layout = ({ children }: Props): JSX.Element => (
  <div className="font-sans font-thin">
    <Header navigations={navigations} />
    {children}
    <Footer />
  </div>
);

export default Layout;
