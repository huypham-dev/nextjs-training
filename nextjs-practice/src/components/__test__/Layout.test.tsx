import { mount, shallow } from "enzyme";
import Footer from "../common/Footer";
import Header from "../common/Header";
import Layout from "../common/Layout";

describe("Layout", () => {
  it("Snapshot", () => {
    const component = shallow(
      <Layout>
        <div>Children</div>
      </Layout>
    );

    expect(component).toMatchSnapshot();
  });

  it("Render correctly", () => {
    const container = mount(
      <Layout>
        <div>Children</div>
      </Layout>
    );

    expect(container.find(Header).length).toBe(1);
    expect(container.find(Footer).length).toBe(1);
  });
});
