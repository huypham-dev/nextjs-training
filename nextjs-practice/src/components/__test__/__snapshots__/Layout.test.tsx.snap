// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshot 1`] = `
<div
  className="font-sans font-thin"
>
  <Memo(Header)
    navigations={
      Array [
        Object {
          "label": "Home",
          "path": "/",
        },
        Object {
          "label": "Blog",
          "path": "/blog",
        },
        Object {
          "label": "Contact",
          "path": "/contact",
        },
      ]
    }
  />
  <div>
    Children
  </div>
  <Memo(Footer) />
</div>
`;
